### ISSUE TITLE
Support specifying data and socket locations at startup

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2549]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2545
- Created at 2015-01-07 11:49:59 by [nkondras](https://pagure.io/user/nkondras)
- Closed at 2020-04-21 14:02:09 as wontfix
- Assigned to nobody
---

Integration testing done within "make check" needs to run under regular users which have no access to the regular build-configured data and socket locations. Build needs to be configured with custom user-accessible locations to enable integration testing.

Currently, these locations are spread all over the binaries as static strings. The mentions of at least some of the paths can be found by running this command in a regularly-configured build directory:

    find -type f -perm -111 -name '*.so*' -or -name 'lt-*' |
        xargs strings -f | grep var/lib/sss

It would be beneficial for sssd, pam_sss, libnss_sss and probably other binaries to support specifying the data and socket locations at startup time. This would allow integration testing to place each instance's data in temporary, user-accessible directories and potentially allow parallel test execution with several sssd instances.

The sssd can receive these locations either through command-line options or through configuration file options. The libraries might need to resort to interpreting environment variables, although this approach might need to be screened for security issues.

### Comments

---

**Comment from [nkondras](https://pagure.io/user/nkondras) at [2015-01-08 16:27:29](https://pagure.io/SSSD/sssd/issue/2545#243158)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD Continuous integration
owner: somebody =>

---

**Comment from [nkondras](https://pagure.io/user/nkondras) at [2015-07-30 16:28:09](https://pagure.io/SSSD/sssd/issue/2545#243159)**

Fields changed

rhbz:  => 0

---

**Comment from [nkondras](https://pagure.io/user/nkondras) at [2015-11-04 14:52:09](https://pagure.io/SSSD/sssd/issue/2545#243160)**

Fields changed

blockedby:  => #2859
sensitive:  => 0

---

**Comment from [nkondras](https://pagure.io/user/nkondras) at [2017-02-24 15:08:12](https://pagure.io/SSSD/sssd/issue/2545#243161)**

**Metadata Update from @nkondras**:
- Issue marked as depending on: #2859
- Issue set to the milestone: SSSD Continuous integration

---

**Comment from [nkondras](https://pagure.io/user/nkondras) at [2020-04-21 14:02:11](https://pagure.io/SSSD/sssd/issue/2545#643288)**

**Metadata Update from @pbrezina**:
- Custom field blocking adjusted to 2859
- Custom field design_review adjusted to on (was: 0)
- Custom field mark adjusted to on (was: 0)
- Custom field patch adjusted to on (was: 0)
- Custom field review adjusted to on (was: 0)
- Custom field sensitive adjusted to on (was: 0)
- Custom field testsupdated adjusted to on (was: 0)
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
